src/loaders/express.js

```diff
@@ -82,7 +82,7 @@ module.exports = async (app) => {
   app.use((req, res, next) => {
     res.set('X-Amzn-Trace-Id', `Root=${AWSXRay.getSegment().trace_id}`);
     AWSXRay.getSegment().addAnnotation('podName', config.podName);
-
+    AWSXRay.getSegment().addAnnotation('userId', req.headers.userid);
```

Comment:
There is already a predetermined way to add user ID stuff:

https://docs.aws.amazon.com/xray/latest/devguide/xray-sdk-nodejs-segment.html#xray-sdk-nodejs-segment-userid

I would use the already existing [middlewares](https://github.com/biomage-ltd/api/blob/develop/src/utils/authMiddlewares.js) for authentication to add this information directly to the X-Ray trace.

You can use [authenticationMiddlewareSocketIO](https://github.com/biomage-ltd/api/blob/develop/src/utils/authMiddlewares.js#L92) to get the claim and from the claim you can get the user ID for Socket.IO requests (which you have not done here).

For normal HTTP requests, you can just add a custom middleware after the Express auth middleware is called, i.e. after [here](https://github.com/biomage-ltd/api/blob/develop/src/loaders/express.js#L92). By that point you should have `req.user` filled in with the required information, so you can just add it to the current segment without any problems.